src/autopas/utils/EnergySensor.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file EnergySensor.h
+ * @author Maximilian Praus
+ * @date 29.06.2024
+ */
+
+#pragma once
+#include "autopas/options/EnergySensorOption.h"
+#ifdef AUTOPAS_ENABLE_ENERGY_MEASUREMENTS
+#include "pmt.h"
+#endif
+
+namespace autopas::utils {
+
+/**
+ * Measure the energy consumption of a simulation on multiple hardwares.
+ * Simulation needs to run on an Linux system to use Power measurement toolkit
+ */
+
+class EnergySensor {
+ public:
+  /**
+   * Constructor for energy sensor. Takes @param sensor as possible energy sensors to use
+   */
+  EnergySensor(EnergySensorOption sensor);
+
+  /**
+   * Initializes the EnergySenor
+   * @param tuningMetricIsEnergy whether energy is selected as the tuning metric
+   * @return whether energy measurement is turned on
+   */
+  bool init(bool tuningMetricIsEnergy);
+
+  /**
+   * Start measurements. Sets _start
+   * @return returns true if _sensor available. Otherwise false
+   */
+  bool startMeasurement();
+
+  /**
+   * Ending measurement. Sets _end
+   * @return returns true if _sensor available. Otherwise false
+   */
+  bool endMeasurement();
+
+  /**
+   * Get joules consumed between start and end state
+   * @return double for watts consumed
+   */
+  double getJoules() const;
+
+  /**
+   * Get watts consumed between start and end state
+   * @return double for watts consumed
+   */
+  double getWatts() const;
+
+  /**
+   * Get seconds between current start and end state
+   * @return double for seconds between time stamps
+   */
+  double getSeconds() const;
```

Comment:
I find the name non-intuitive. Is maybe something like `getEnergyDeltaT()` a better name?

- [ ] This also affects the corresponding column name in the iteration logger